Kommunicate/Classes/KMConversationViewController.swift

```diff
@@ -166,6 +166,36 @@ open class KMConversationViewController: ALKConversationViewController {
         }
     }
 
+    open override func addMessagesToList(_ messageList: [Any]) {
+        guard let messages = messageList as? [ALMessage] else { return }
+        var filteredArray = [ALMessage]()
+        let contactService = ALContactService()
+        for message in messages {
+            if viewModel.channelKey != nil, viewModel.channelKey == message.groupId {
+                let delayInterval = KMAppUserDefaultHandler.shared.botMessageDelayInterval
+                if delayInterval > 0 {
+                    let alContact = contactService.loadContact(byKey: "userId", value:  message.to)
+                    if alContact?.roleType == NSNumber.init(value: AL_BOT.rawValue) {
+                        updateTyingStatus(status: true, userId: message.to)
+                        DispatchQueue.main.asyncAfter(deadline: .now() + DispatchTimeInterval.milliseconds(delayInterval)) {
+                            self.viewModel.addMessagesToList([message])
+                        }
+                    } else {
+                        filteredArray.append(message)
+                    }
+                } else {
+                    filteredArray.append(message)
+                }
+            } else if message.channelKey == nil, viewModel.channelKey == nil, viewModel.contactId == message.to {
```

Comment:
`else` will be sufficient as we have to add messages in all other cases. We can remove the conditional check.